src/common/reducers/entities/index.js

```diff
@@ -1,26 +1,22 @@
 import merge from 'lodash/merge';
 
 import * as RepoActionTypes from '../../actions/repository';
-
 import repository from './repository';
 
-
 // TODO snaps
 
-export function entities(state = { snaps: {}, repos: {} }, action) {
+export function entities(state = {
+  snaps: {},
+  repos: {},
+  owners: {}
+}, action) {
   if (action.payload && action.payload.entities) {
     return merge({}, state, action.payload.entities);
   }
 
   // only modify repos if action is one of REPO_ types
-  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
-    return {
-      ...state,
-      repos: {
-        ...state.repos,
-        [action.payload.id]: repository(state.repos[action.payload.id], action)
-      }
-    };
+  if (RepoActionTypes[action.type]) {
+    return repository(state, action);
```

Comment:
11:48 AM <beowulf> bartaz: ok, i see your point now
11:48 AM <beowulf> bartaz: so ... right, repository as it currently is should be back in entities ... yeah, i get you
11:49 AM <beowulf> and _repository is what stays (because it deals with a repo, not an entity collection)
11:49 AM <beowulf> bartaz: have i understood that correctly?
11:50 AM <bartaz> yes - and actually if 'reposEntities' (or howerer we call it - 'repos' maybe) would be implemented in a way that don't knows about entities, I'd be ok with having it in separate file
11:51 AM <bartaz> like 'repos.js' that would have repos and repository
11:51 AM <beowulf> bartaz: in a future where snaps are also involved, the repository func (that returns and entity collect) would be a helper function that you would pass 'snap' or 'repo' too
11:51 AM <bartaz> right, the only difference would be the prop name
11:51 AM <beowulf> bartaz: i think it has to know about entities, no? it's really just the mapping ... yeah
11:52 AM <beowulf> so it should be in entities, and then the naming isn't so much of a problem either
11:52 AM <bartaz> yes
11:52 AM <beowulf> it really has nothign to do with repos
11:52 AM <beowulf> it's just a path finder